### ISSUE TITLE
Create horizonclient release 3.0 once Horizon 1.0 has been merged into master

### ISSUE THREAD

**Issue by abuiles** [stellar/go#2293]
We should cherry-pick the following commits into it:

https://github.com/stellar/go/commit/ef52dffcb6600523ef410d59790000df03b49866 
https://github.com/stellar/go/commit/b973a23b767241b6d2a7fb3196c245e7ec96a004
https://github.com/stellar/go/commit/ababa11f23459aa6e7e7e8dee46778d49b9ee829

**Comment by ire-and-curses** [comment]
For context: this is because we are releasing Horizon 2.1 with backwards compatibility, which means it depends on the older version of Horizon's protocols file. We're maintaining backwards compatibility because it would be unexpected for a user of the client to discover that their code broke when they upgraded on a minor release.

Once Horizon 1.0 is merged back to master, it's possible to create the new (breaking) 3.0 release branch for horizonclient, where we finally discontinue the pre-1.0 support. At that point, we need to pick up these commits from the divergent 2.1 branch to maintain functionality.

We'll plan to release this major release to sync with the upcoming stellar-core protocol 13, since that will likely force additional breaking changes to the SDK.

**Comment by abuiles** [comment]
Done! 

The branch is here https://github.com/stellar/go/tree/release-horizonclient-v3.0.0 and it has all the relevant commits applied to it.
